### ISSUE TITLE
Volume cannot be mounted: can't read superblock on /dev/loop0

### ISSUE THREAD

**Issue by papanito** [kadalu/kadalu#196]
I get the following error with `kadalu 0.5.2`

```
6m4s        Warning   FailedMount              pod/nextcloud-mariadb-slave-0                            MountVolume.SetUp failed for volume "pvc-56965f29-7886-48b8-8751-1063c8d41624" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = Exception calling application: [32]  mount: /var/lib/kubelet/pods/7ecab6b0-d47c-4ab1-9edd-9d5b3364fa19/volumes/kubernetes.io~csi/pvc-56965f29-7886-48b8-8751-1063c8d41624/mount: wrong fs type, bad option, bad superblock on /dev/loop0, missing codepage or helper program, or other error.
6m2s        Warning   FailedMount              pod/nextcloud-mariadb-slave-0                            MountVolume.SetUp failed for volume "pvc-56965f29-7886-48b8-8751-1063c8d41624" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = Exception calling application: [32]  mount: /var/lib/kubelet/pods/7ecab6b0-d47c-4ab1-9edd-9d5b3364fa19/volumes/kubernetes.io~csi/pvc-56965f29-7886-48b8-8751-1063c8d41624/mount: /mnt/storage-pool-main/virtblock/06/f9/pvc-56965f29-7886-48b8-8751-1063c8d41624 is not a block device; try "-o loop".
113s        Warning   FailedMount              pod/nextcloud-mariadb-slave-0                            MountVolume.SetUp failed for volume "pvc-56965f29-7886-48b8-8751-1063c8d41624" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = Exception calling application: [32]  mount: /var/lib/kubelet/pods/7ecab6b0-d47c-4ab1-9edd-9d5b3364fa19/volumes/kubernetes.io~csi/pvc-56965f29-7886-48b8-8751-1063c8d41624/mount: can't read superblock on /dev/loop0.
```

However not sure if this is really an issue

**Steps to reproduce**

1. Cleanup all using the cleanup script

1. Re-deploy kadalu

1. Create `storage-config.yaml`

   ```yaml
    apiVersion: kadalu-operator.storage/v1alpha1
    kind: KadaluStorage
    metadata:
    # This will be used as name of PV Hosting Volume
      name: storage-pool-main
    spec:
      type: Replica3
      storage:
        - node: node001
          device: /dev/md3
        - node: node002
          device: /dev/md3
        - node: node003
          device: /dev/md3
   ```
1. Deploy nextcloud

   ```bash
   helm install nextcloud stable/nextcloud --namespace nextcloud -f nextcloud.values.yml
   ```

**Expected Result**

No errors

**Actual Result** 

```
kubectl -nnextcloud get events 
...
6m4s        Warning   FailedMount              pod/nextcloud-mariadb-slave-0                            MountVolume.SetUp failed for volume "pvc-56965f29-7886-48b8-8751-1063c8d41624" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = Exception calling application: [32]  mount: /var/lib/kubelet/pods/7ecab6b0-d47c-4ab1-9edd-9d5b3364fa19/volumes/kubernetes.io~csi/pvc-56965f29-7886-48b8-8751-1063c8d41624/mount: wrong fs type, bad option, bad superblock on /dev/loop0, missing codepage or helper program, or other error.
6m2s        Warning   FailedMount              pod/nextcloud-mariadb-slave-0                            MountVolume.SetUp failed for volume "pvc-56965f29-7886-48b8-8751-1063c8d41624" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = Exception calling application: [32]  mount: /var/lib/kubelet/pods/7ecab6b0-d47c-4ab1-9edd-9d5b3364fa19/volumes/kubernetes.io~csi/pvc-56965f29-7886-48b8-8751-1063c8d41624/mount: /mnt/storage-pool-main/virtblock/06/f9/pvc-56965f29-7886-48b8-8751-1063c8d41624 is not a block device; try "-o loop".
113s        Warning   FailedMount              pod/nextcloud-mariadb-slave-0                            MountVolume.SetUp failed for volume "pvc-56965f29-7886-48b8-8751-1063c8d41624" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = Exception calling application: [32]  mount: /var/lib/kubelet/pods/7ecab6b0-d47c-4ab1-9edd-9d5b3364fa19/volumes/kubernetes.io~csi/pvc-56965f29-7886-48b8-8751-1063c8d41624/mount: can't read superblock on /dev/loop0.
```

Volume claims seem fine

```
kubectl -n nextcloud get pvc
NAME                              STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
data-nextcloud-mariadb-master-0   Bound    pvc-0a9d3e07-36a9-4e8f-9ae0-87a448d2ea7d   8Gi        RWO            kadalu.replica3   4m34s
data-nextcloud-mariadb-slave-0    Bound    pvc-56965f29-7886-48b8-8751-1063c8d41624   8Gi        RWO            kadalu.replica3   4m34s
nextcloud-nextcloud               Bound    pvc-32529709-29f2-41a9-ba65-16306066ad21   500Gi      RWO            kadalu.replica3   4m35s
```

[csi-nodeplugin-9ggpw.log](https://github.com/kadalu/kadalu/files/4270335/csi-nodeplugin-9ggpw.log)
[csi-provisioner-0.log](https://github.com/kadalu/kadalu/files/4270336/csi-provisioner-0.log)
[nextcloude-events.log](https://github.com/kadalu/kadalu/files/4270337/nextcloude-events.log)
[server-storage-pool-main-0-node001-0.log](https://github.com/kadalu/kadalu/files/4270338/server-storage-pool-main-0-node001-0.log)

**Additional info**

N/A

**Comment by amarts** [comment]
We have noticed 2 things here.

1. usage of `/dev/loopNN`. The stability of this approach depends on host machine OS/version too.
2. RWO doesn't mean we have to provide a 'Block' storage, it means, the PV provided should be used by just one application.

With that in mind, we did add a fix (#174), which should help in reducing the issues caused by RWO usecases.

This should be available only with 0.6.0 release, and we noticed that there was an issue with tag label 'latest', and hence you hit the issue even with latest release. With latest master, we have fixed the issue with 'kadalu-operator.yaml' file, so it should be fine now. Let us know if you use latest operator and latest CSI drivers, what happens in this PV request.
